### PR TITLE
Clean branch

### PR DESCRIPTION
**Pull request by someaka**
# StreamingSectors and stuff

<PULL REQUEST TITLE>

Implemented:
- added Copy Paste and Delete selection
- added export for nodeData
- added imports for Objects Spawner and Unreal formats
- added option to import gltf with no original (WIP)
- added an option to import boneRigMatrices for meshes
- added an option to import rigs (very very WIP)
- added a refresh file button for streamingsectors
- probably more stuff i forgot

Fixed:
- YES

<Additional notes>

🍌 🍐 🍓 🍎 🍏 🍒 🍑

### CODE REVIEW

File: WolvenKit.App/ViewModels/Documents/MeshPreviewBlock.cs [hunk 1]
```diff
@@ -313,7 +313,11 @@ public void OnMouseDown3DHandler(object sender, MouseDown3DEventArgs e)
                     var sector = Sectors.Where(x => x.Text == text).FirstOrDefault();
                     if (sector != null)
                     {
-                        LoadSector(sector);
+                        try
+                        {
+                            LoadSector(sector);
+                        }
+                        catch { }
```
Comment (seberoth) [inline WolvenKit.App/ViewModels/Documents/MeshPreviewBlock.cs:320]: Please add logging to the catch (also applies to the other try-catches)

File: WolvenKit.Common/RED4/CR2W/JSON/PrimitiveConverter.cs [hunk 2]
```diff
@@ -2348,7 +2352,17 @@ public RedClassConverter(ReferenceResolver<RedBaseClass> referenceResolver)
...
+                        /*
+                                                static string? FirstCharToLowerCase( string? str)
+                                                {
+                                                    if (!string.IsNullOrEmpty(str) && char.IsUpper(str[0]))
+                                                        return str.Length == 1 ? char.ToLower(str[0]).ToString() : char.ToLower(str[0]) + str[1..];
+
+                                                    return str;
+                                                }
+                        */
+                        //what's the difference between RedName and Name ?!
```
Comment (seberoth) [inline WolvenKit.Common/RED4/CR2W/JSON/PrimitiveConverter.cs:2364]: RedName is CDPRs RTTI name, Name is our C# Name (mostly just first char uppercase but there are exceptions)

File: WolvenKit.Common/RED4/CR2W/JSON/PrimitiveConverter.cs [hunk 3]
```diff
@@ -2602,10 +2631,19 @@ public void End()
...
     {
+        try
+        {
+            ReferenceIdToObjectMap[referenceId] = value;
+        }
+        catch (Exception ex)
+        {
+            throw new JsonException($"{ex}");
+        }
+/*
```
Comment (seberoth) [inline WolvenKit.Common/RED4/CR2W/JSON/PrimitiveConverter.cs:2642]: Why? The value shouldn't be changed after its set, so TryAdd is better.
If its changed probably something other is broken.

File: WolvenKit.Modkit/RED4/Import.cs [hunk 4]
```diff
@@ -633,14 +637,31 @@ void SetTextureGroupSetup(STextureGroupSetup setup, CR2WFile cr2w)
 
         private bool ImportGltf(RedRelativePath rawRelative, DirectoryInfo outDir, GltfImportArgs args)
         {
+            var am = Locator.Current.GetService<IArchiveManager>();
+            var somefile = am.Lookup(11327483717126307343); //some ice cube  11327483717126307343
+                                                            //some cube     15966709641681476693
```
Comment (seberoth) [inline WolvenKit.Modkit/RED4/Import.cs:642]: ???

Comment (someaka) [reply WolvenKit.Modkit/RED4/Import.cs:642]: uses the ice cube mesh as default if no mesh is provided, planning to polish that feature a bit more

File: WolvenKit.RED4.Archive/IO/PackageReader.cs [hunk 5]
```diff
@@ -60,7 +60,8 @@ public RedBaseClass ReadClass(Type type)
                     if (fieldType != prop.Type)
                     {
                         var propName = $"{RedReflection.GetRedTypeFromCSType(instance.GetType())}.{varName}";
-                        throw new InvalidRTTIException(propName, prop.Type, fieldType);
+                        //throw new InvalidRTTIException(propName, prop.Type, fieldType);
+
```
Comment (seberoth) [inline WolvenKit.RED4.Archive/IO/PackageReader.cs:64]: Revert please

File: Tests/WolvenKit.FunctionalTests/appsettings.json [hunk 6]
```diff
@@ -1,5 +1,5 @@
 {
-  "GameDirectory": "Z:\\GOG Galaxy\\Games\\Cyberpunk 2077",
+  "GameDirectory": "J:\\Cyberpunk 2077",
   "WriteToFile": true,
```
Comment (seberoth) [inline Tests/WolvenKit.FunctionalTests/appsettings.json:3]: Revert please

File: Tests/WolvenKit.Utility/appsettings.json [hunk 7]
```diff
@@ -1,5 +1,5 @@
 {
-  "GameDirectory": "Z:\\GOG Galaxy\\Games\\Cyberpunk 2077",
+  "GameDirectory": "J:\\Cyberpunk 2077",
```
Comment (seberoth) [inline Tests/WolvenKit.Utility/appsettings.json:2]: Revert please

File: WolvenKit.Common/RED4/CR2W/JSON/PrimitiveConverter.cs (hunk 3 above)
Comment (someaka) [reply WolvenKit.Common/RED4/CR2W/JSON/PrimitiveConverter.cs:2642]: tried a bunch of things on that one, it seems the issue is with duplicates,
that's the only thing i could get working somehow

File: WolvenKit/Views/Tools/ProjectExplorerView.xaml [hunk 8]
```diff
@@ -145,6 +145,16 @@
                             Kind="Pencil" />
                     </MenuItem.Icon>
                 </MenuItem>
+                <!--<MenuItem Command="{Binding TeststuffCommand}" Header="TestStuff">
```
Comment (rfuzzo) [inline WolvenKit/Views/Tools/ProjectExplorerView.xaml:148]: Can you delete this?

File: WolvenKit.App/ViewModels/Shell/SomeUtils.cs [hunk 9]
```diff
@@ -0,0 +1,857 @@
+using System;
```
Comment (rfuzzo) [inline WolvenKit.App/ViewModels/Shell/SomeUtils.cs:1]: Can this get a more descriptive name than "SomeUtils" ?

File: WolvenKit.App/ViewModels/Tools/PropertiesViewModel.cs [hunk 10]
```diff
@@ -347,19 +347,27 @@ public async Task LoadImageFromStream(Stream stream)
...
+                    bounds = SharpDX.BoundingBox.Merge(bounds, mesh.BoundsWithTransform);
+                }
+
+                ModelGroupBounds = bounds;
+                ModelGroup.Reset(meshes);
+            }
+            catch
             {
-                bounds = SharpDX.BoundingBox.Merge(bounds, mesh.BoundsWithTransform);
+
```
Comment (rfuzzo) [inline WolvenKit.App/ViewModels/Tools/PropertiesViewModel.cs:368]: log the error?

**Review by rfuzzo — CHANGES_REQUESTED** [review]

File: WolvenKit.Common/RED4/CR2W/JSON/PrimitiveConverter.cs (hunk 3 above)
Comment (seberoth) [reply WolvenKit.Common/RED4/CR2W/JSON/PrimitiveConverter.cs:2642]: RefId/BufferId should never be duplicated. So either the Writer is broken or its an user error (modified by hand). Anyway that shouldn't be solved here

File: WolvenKit.App/ViewModels/Tools/PropertiesViewModel.cs (hunk 10 above)
Comment (someaka) [reply WolvenKit.App/ViewModels/Tools/PropertiesViewModel.cs:368]: uhhh ... pretty sure that's not something i wrote, that's concerning

**Review by rfuzzo — APPROVED** [review]
lgtm

**Pull request merged**